src/control/positionControl/PositionControl.cpp

```diff
@@ -0,0 +1,53 @@
+//
+// Created by ratoone on 18-11-19.
+//
+
+#include "control/positionControl/PositionControl.h"
+#include <interface/api/Input.h>
+#include <control/positionControl/pathTracking/NumTreesTracking.h>
+
+namespace rtt::ai::control {
+
+PositionControl::PositionControl(const std::vector<rtt::world_new::robot::Robot> &robots): robots(robots) {
+    collisionDetector = new CollisionDetector(robots);
+    pathPlanningAlgorithm = new NumTreesPlanning(*collisionDetector);
+    pathTrackingAlgorithm = new NumTreesTracking();
+}
+
+//TODO: add projection to outside defence area (project target position)(is this really needed?)
+RobotCommand
+PositionControl::computeAndTrackPath(world::Field *field, int robotId, const Vector2 &currentPosition,
+                                     const Vector2 &currentVelocity, const Vector2 &targetPosition) {
+    //TODO: this is a workaround caused by the fact that the field is not global
+    collisionDetector->setField(field);
```

Comment:
What? Pls don't use raw pointers but copy by reference or just copy if you can because this might cause UB at any point in time.